### PR TITLE
part_7

### PR DESCRIPTION
**Pull request by Baldinini**

### CODE REVIEW

File: src/main/java/mate/academy/config/AppConfig.java [hunk 1]
```diff
@@ -15,7 +15,8 @@
 @PropertySource("classpath:db.properties")
 @ComponentScan(basePackages = {
         "mate.academy.service",
-        "mate.academy.dao"
+        "mate.academy.dao",
+        "mate.academy.service.mapper"
```
Comment (yurwing) [inline src/main/java/mate/academy/config/AppConfig.java]: `        "mate.academy.service.mapper"
`
I think it's redundant

File: src/main/java/mate/academy/service/mapper/impl/MovieSessionMapperImpl.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import mate.academy.model.dto.MovieSessionRequestDto;
+import mate.academy.model.dto.MovieSessionResponseDto;
+import mate.academy.service.CinemaHallService;
+import mate.academy.service.MovieService;
+import mate.academy.service.mapper.MovieSessionMapper;
+import org.springframework.stereotype.Component;
+
+@Component
+public class MovieSessionMapperImpl implements MovieSessionMapper {
+    public static final String FORMATTER = "dd.MM.yyyy HH:mm";
```
Comment (yurwing) [inline src/main/java/mate/academy/service/mapper/impl/MovieSessionMapperImpl.java]:
```suggestion
    private static final String FORMATTER = "dd.MM.yyyy HH:mm";
```

**Review by yurwing — APPROVED** [review]

File: src/main/java/mate/academy/controllers/CinemaHallController.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/cinema-halls")
+public class CinemaHallController {
+    private final CinemaHallService cinemaHallService;
+    private final CinemaHallMapper cinemaHallMapper;
+
+    @Autowired
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/controllers/CinemaHallController.java:23]: let's use the same approach everywhere - either use this annotation over the constructor - or not

File: src/main/java/mate/academy/model/dto/CinemaHallRequestDto.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
+package mate.academy.model.dto;
+
+public class CinemaHallRequestDto {
+    private Long id;
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/model/dto/CinemaHallRequestDto.java]: do we need this here?

File: src/main/java/mate/academy/model/dto/MovieRequestDto.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
+package mate.academy.model.dto;
+
+public class MovieRequestDto {
+    private Long id;
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/model/dto/MovieRequestDto.java]: same

File: src/main/java/mate/academy/service/mapper/GenericMapper.java [hunk 6]
```diff
@@ -0,0 +1,7 @@
+package mate.academy.service.mapper;
+
+public interface GenericMapper<I,T,K> {
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/service/mapper/GenericMapper.java]:
```suggestion
public interface GenericMapper<I, T, K> {
```

File: src/main/java/mate/academy/service/mapper/GenericMapper.java [hunk 7]
```diff
@@ -0,0 +1,7 @@
...
+    I convertToDto(T t);
+
+    T convertToEntity(K k);
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/service/mapper/GenericMapper.java]:
```suggestion
    I convertToDto(T entity);

    T convertToEntity(K requestDto);
```

File: src/main/java/mate/academy/service/mapper/impl/CinemaHallMapperImpl.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+        cinemaHallDto.setId(cinemaHall.getId());
+        cinemaHallDto.setCapacity(cinemaHall.getCapacity());
+        cinemaHallDto.setDescription(cinemaHall.getDescription());
+        return cinemaHallDto;
+    }
+
+    @Override
+    public CinemaHall convertToEntity(CinemaHallRequestDto cinemaHallRequestDto) {
+        CinemaHall cinemaHall = new CinemaHall();
+        cinemaHall.setId(cinemaHallRequestDto.getId());
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/service/mapper/impl/CinemaHallMapperImpl.java]: do we need to set id?

File: src/main/java/mate/academy/service/mapper/impl/MovieMapperImpl.java [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+        movieDto.setId(movie.getId());
+        movieDto.setTitle(movie.getTitle());
+        movieDto.setDescription(movie.getDescription());
+        return movieDto;
+    }
+
+    @Override
+    public Movie convertToEntity(MovieRequestDto movieRequestDto) {
+        Movie movie = new Movie();
+        movie.setId(movieRequestDto.getId());
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/service/mapper/impl/MovieMapperImpl.java]: same

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**